content/posts/why-meson/index.md

```diff
@@ -0,0 +1,174 @@
++++
+"author" = "ret2libc"
+"title" = "Why meson"
+"date" = "2020-12-11"
+"summary" = "Why we chose meson"
+"tags" = ["dev"]
+
+"ShowToc" = false
+"weight" = 2
++++
+
+# TL;DR
+- It is fast, no files are recompiled if not necessary
+- It is declarative, easy to understand, approach
+- It keeps your source directory clean with out-of-source builds
+- It makes it easy to build and run multiple versions of Rizin
+
+# A bit of context
+
+Historically radare2 has been compiled with the usual `./configure;make`
+approach. This essentially consists of a sh script, `configure`, and a (set
+of) Makefile(s). `configure` allows the user to customize the
+compilation/installation process performed by `make`, by setting for example
+the destination directories where executables, libraries, etc. are
+installed on the system. It is also used to enable/disable specific features
+(e.g. the debugger) or check for the existence of specific libraries, header
+files, functions, compiler or linker arguments.
+
+To some, this may be very similar to what is done by Autotools, however in
+radare2/Rizin case, `configure` is generated by another sh script, `acr`, by
+parsing a `configure.acr` file. `acr` is a tool developed by pancake, the
+author of radare2, and it is an autoconf replacement.
+
+In 2017 pancake started introducing another build system which could be used
```

Comment:
I think it worth to mention another attempts to try different build systems:
- [Jam](https://en.wikipedia.org/wiki/Perforce_Jam) - https://github.com/radareorg/radare2/commit/dba63fbd3ca304eec229a510322af658b8a04a0f
- [Node.js-based custom build system](https://github.com/radareorg/radare2/commit/11ca51269f206819ffdee472a5264edae7ef138d)